### DISCUSSION TITLE
using sqlalchemy class models to generate queries

### DISCUSSION THREAD

**Discussion by nbraun-wolf** [encode/databases#433] [Q&A] [answered]
Hi, I am wondering how sqlalchemy model classes could be used to get a query to execute. In the example, the Table class is used.

```python
notes = sqlalchemy.Table(...)
query = notes.insert()
values = {"text": "example1", "completed": True}
await database.execute(query=query, values=values)
```

But I am looking to do something like this from my model classes.

```python
class User(Model):
    ...
```

Is there a way of doing that?

**Comment by aminalaee** [comment, marked as answer]
Hey, I think with SQLAlchemy 1.4 this should work:

```python
from sqlalchemy import select

query = select(User).where(User.name == 'spongebob')
await database.execute(query)
```

**Reply by nbraun-wolf** [reply]
Yes that works. Thank you :) We just need to convert query to string it seems.

```python
await database.execute(str(query))
```
